### ISSUE TITLE
Custom $select returning `id`

### ISSUE THREAD

**Issue by Mattchewone** [feathersjs-ecosystem/feathers-memory#58]
### Steps to reproduce
```js
      return people.find({
        query: {
          name: 'Tester',
          $select: ['name']
        }
      }).then(person => {
        assert.deepEqual(person[0], { name: 'Tester' }, 'deepEquals the same');
      });
```

### Expected behavior
I would expect the returned data to only have the properties I specified in the `$select`.
Expected:
```js
{
  name: 'Tester'
}
```

### Actual behavior
The `id` field is added to the return.
Actual:
```js
{
  id: 1,
  name: 'Tester'
}
```

### Solution
This seems to be coming from replacing the `filters.$select` from a custom `_.pick` to using the `feathers-commons` select method here - https://github.com/feathersjs-ecosystem/feathers-memory/commit/4dcc6080a069f35bfe72ab656fe1cdfd6c525ad4#diff-1fdf421c05c1140f6d71444ea2b27638L41
If we remove `this.id` from here: `const map = select(params, this.id);` this should remedy the problem and give `feathers-memory` the same behaviour as the other database adapters.

### System configuration
`Node 8.4`
`feathers-memory@latest`

**Comment by marshallswain** [comment]
Wouldn't you always want the id field?  How would you modify any record, otherwise?

**Comment by Mattchewone** [comment]
Well this would mean that the behaviour would be different to the db adapters that use a `select` and would not automagically add the `idField` to the selected result.

**Comment by marshallswain** [comment]
Good point.  So you'd just need to include `_id` in the select statement.

**Comment by Mattchewone** [comment]
Yes in the db adapters you would have to specify `id` if you wished for it to be returned.

**Comment by marshallswain** [comment]
Not to fill the issue up but I am pretty sure that is what is happening here - https://github.com/feathersjs-ecosystem/feathers-mongoose/blob/master/src/service.js#L54

and here - https://github.com/feathersjs-ecosystem/feathers-sequelize/blob/master/src/index.js#L55

As they are using selects as part of the query the `feathers-commons` select is not being used and as such the `id` field is not automagically added.


[10:36] 
Those above are all on `_find`

But weirdly in `feathers-sequelize` the `get` method uses the `feathers-commons` select method which means the behaviour on the get would be different and would return the `id` although not sure you can pass `$select` to get?

https://github.com/feathersjs-ecosystem/feathers-sequelize/blob/master/src/index.js#L113

**Comment by daffl** [comment]
Fixed in v1.3.1
